### PR TITLE
Rust add kerberos 03

### PR DESCRIPTION
**Pull request by chifflier**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Describe changes:
- Update of #3345, #3371 (rebase + rewrite)
- Rename detection keywords `krb5.*` to `krb5_*`
- Use sticky buffers (multiple) for `krb5_cname` and `krb5_sname`

Examples of rules (also in documentation):
```
alert krb5 any any -> any any (msg:"Kerberos 5 AS-REQ message"; krb5_msg_type:10; sid:12345678; rev:1;)
alert krb5 any any -> any any (msg:"Kerberos 5 krbtgt sname"; krb5_sname; content:"krbtgt"; sid:12345679; rev:1;)
alert krb5 any any -> any any (msg:"Kerberos 5 des cname"; krb5_cname; content:"des"; sid:12345680; rev:1;)
alert krb5 any any -> any any (msg:"Kerberos 5 ERR-TYPE-NOTSUPP"; krb5_err_code:14; sid:12345681; rev:1;)
```

### CODE REVIEW

**Comment by pevma** [comment]
I just had a short quick look  - have an observation - seems the keywords/app layer availability do not show up when queried  
```
pevma@DonPedro:~/Work/Suricata/tests/suricata (rust-add-kerberos-03)$ sudo /opt/suritest/bin/suricata --list-app-layer-protos |grep krb
pevma@DonPedro:~/Work/Suricata/tests/suricata (rust-add-kerberos-03)$ sudo /opt/suritest/bin/suricata --list-keywords |grep krb
pevma@DonPedro:~/Work/Suricata/tests/suricata (rust-add-kerberos-03)$ 
```
The documentation is quite nice with examples about all krb5 keyword possibilities!

**Comment by chifflier** [comment]
Hi @pevma , are you sure you tested the correct version ? In the build directory, the keywords are listed:
```
$ ./src/suricata --list-app-layer-protos |grep krb
krb5
$ ./src/suricata --list-keywords |grep krb
- krb5_err_code
- krb5_msg_type
- krb5_cname
- krb5_sname
```

**Comment by pevma** [comment]
I have compiled the first time without rust - but then when I corrected (sorry) that I have the following:

```
suricata (rust-add-kerberos-03)$ sudo /opt/suritest/bin/suricata -V
This is Suricata version 4.1.0-dev (rev 729aef0c)
suricata (rust-add-kerberos-03)$ sudo /opt/suritest/bin/suricata --build-info |grep Rust
  Rust support (experimental):             yes
  Rust strict mode:                        no
  Rust debug mode:                         no
suricata (rust-add-kerberos-03)$ sudo /opt/suritest/bin/suricata --list-keywords |grep -i krb5
- krb5_err_code
- krb5_msg_type
- krb5_cname
- krb5_sname
suricata (rust-add-kerberos-03)$ sudo /opt/suritest/bin/suricata --list-app-layer-protos |grep krb
suricata (rust-add-kerberos-03)$ 

```
If you can not reproduce it - then is most likely something I missed.

File: suricata.yaml.in [hunk 1]
```diff
@@ -192,6 +193,7 @@ outputs:
         header: X-Forwarded-For
 
       types:
+        - krb5
```
Comment (victorjulien) [inline suricata.yaml.in]: this will lead to error if no rust is enabled. Also, please move it down to be around the nfs/smb ones.

**Comment by chifflier** [comment]
Replaced by #3385 (merging fixes for all comments)

**Pull request closed without merging**